repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -1026,17 +1027,30 @@ class SparkILoop(
 
   @DeveloperApi
   def createSQLContext(): SQLContext = {
-    val name = "org.apache.spark.sql.hive.HiveContext"
+    useHiveContext = sparkContext.getConf.getBoolean("spark.sql.useHiveContext", true)
+    val name = {
+      if (useHiveContext) "org.apache.spark.sql.hive.HiveContext"
+      else "org.apache.spark.sql.SQLContext"
+    }
+
     val loader = Utils.getContextOrSparkClassLoader
     try {
       sqlContext = loader.loadClass(name).getConstructor(classOf[SparkContext])
         .newInstance(sparkContext).asInstanceOf[SQLContext]
-      logInfo("Created sql context (with Hive support)..")
+      if (useHiveContext) {
+        logInfo("Created sql context (with Hive support). To use sqlContext (without Hive), " +
+          "set spark.sql.useHiveContext to false before launching spark-shell.")
+      }
+      else {
+        logInfo("Created sql context.")
+      }
     }
     catch {
-      case _: java.lang.ClassNotFoundException | _: java.lang.NoClassDefFoundError =>
+      case _: java.lang.ClassNotFoundException | _: java.lang.NoClassDefFoundError
+        if useHiveContext =>
         sqlContext = new SQLContext(sparkContext)
-        logInfo("Created sql context..")
+        logInfo("Created sql context without Hive support, " +
+          "build Spark with -Phive to enable Hive support.")
```

Comment:
It is possible that `-Phive` is used but there was an exception. Can we also update the log message for this case?